core/database/src/main/java/com/pronoidsoftware/core/database/RoomLocalAgendaDataSource.kt

```diff
@@ -113,11 +119,97 @@ class RoomLocalAgendaDataSource @Inject constructor(
         agendaDao.deleteAllTasks()
     }
 
+    // Events
+    override fun getAllEvents(): Flow<List<AgendaItem.Event>> {
+        return agendaDao.getAllEvents()
+            .map { eventEntities ->
+                eventEntities.map { it.toEvent() }
+            }
+    }
+
+    override fun getEventsForDate(targetDate: String): Flow<List<AgendaItem.Event>> {
+        return agendaDao.getEventsForDate(targetDate)
+            .map { eventEntities ->
+                eventEntities.map { it.toEvent() }
+            }
+    }
+
+    override suspend fun upsertEvent(event: AgendaItem.Event): Result<EventId, DataError.Local> {
+        return try {
+            val entity = event.toEventEntity()
+            agendaDao.upsertEvent(entity)
+            agendaDao.upsertAttendees(
+                event.attendees
+                    .map {
+                        it.toAttendeeEntity(
+                            eventId = event.id ?: UUID.randomUUID().toString(),
+                        )
+                    },
+            )
+            agendaDao.upsertPhotos(
+                event.photos
+                    .map {
+                        it.toPhotoEntity(
+                            eventId = event.id ?: UUID.randomUUID().toString(),
+                        )
+                    },
+            )
```

Comment:
Similar as in the last PR, this might better be done in a transaction to avoid inconsistent states where an event is successfully inserted, but its attendees and photos are not (same for the other functions in this class)